aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,709 @@
+			},
+		},
+	})
+}
+
+func testAccCheckSagemakerModelDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_sagemaker_model" {
+			continue
+		}
+
+		resp, err := conn.ListModels(&sagemaker.ListModelsInput{
+			NameContains: aws.String(rs.Primary.ID),
+		})
+		if err == nil {
+			if len(resp.Models) > 0 {
+				return fmt.Errorf("Sagemaker models still exist")
+			}
+
+			return nil
+		}
+
+		sagemakerErr, ok := err.(awserr.Error)
+		if !ok {
+			return err
+		}
+		if sagemakerErr.Code() != "ResourceNotFound" {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckSagemakerModelExists(n string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[n]
+		if !ok {
+			return fmt.Errorf("Not found: %s", n)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("No sagmaker model ID is set")
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).sagemakerconn
+		DescribeModelOpts := &sagemaker.DescribeModelInput{
+			ModelName: aws.String(rs.Primary.ID),
+		}
+		_, err := conn.DescribeModel(DescribeModelOpts)
+		if err != nil {
+			return err
+		}
+
+		return nil
+	}
+}
+
+func testAccSagemakerModelConfig(rName string, image string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-%s"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+`, rName, image, rName)
+}
+
+func testAccSagemakerModelConfigTags(rName string, image string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+	}
+
+	tags {
+		foo = "bar"
+	}
+}
+
+resource "aws_iam_role" "foo" {
+  name = "terraform-testacc-sagemaker-model-%s"
+  path = "/"
+  assume_role_policy = "${data.aws_iam_policy_document.assume_role.json}"
+}
+
+data "aws_iam_policy_document" "assume_role" {
+  statement {
+    actions = [ "sts:AssumeRole" ]
+    principals {
+      type = "Service"
+      identifiers = [ "sagemaker.amazonaws.com" ]
+    }
+  }
+}
+`, rName, image, rName)
+}
+
+func testAccSagemakerModelConfigTagsUpdate(rName string, image string) string {
+	return fmt.Sprintf(`
+resource "aws_sagemaker_model" "foo" {
+	name = "terraform-testacc-sagemaker-model-%s"
+	execution_role_arn = "${aws_iam_role.foo.arn}"
+
+	primary_container {
+		image = "%s"
+	}
+
+	tags {
```

Comment:
An equals sign will be required here in Terraform 0.12 and is backwards compatible with Terraform 0.11.  For operators, this configuration update will be detected with the `terraform 0.12upgrade` command.

```hcl
  tags = {
```